src/www/Inset.ts

```diff
@@ -0,0 +1,266 @@
+/*
+   Copyright 2022 Total Pave Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+export interface IInsetAPI {
+    addListener: (callback: IInsetCallbackFunc) => void;
+    removeListener: (callback: IInsetCallbackFunc) => void;
+    getInsets: () => IInset;
+}
+
+declare global {
+    interface ITotalpave {
+        Insets: IInsetAPI;
+    }
+    interface Window {
+        totalpave: ITotalpave;
+    }
+}
+
+export const SERVICE_NAME: string = "Insets";
+
+export interface IInset {
+    top: number;
+    right: number;
+    bottom: number;
+    left: number;
+}
+
+interface IInsetEvent<T = unknown> {
+    type: 'init' | 'update';
+    data: T;
+}
+
+type IInsetInitEvent = IInsetEvent<string>;
+interface IInsetUpdateEvent extends IInsetEvent<IInset> {
+    id: string;
+}
+
+export type IInsetCallbackFunc = (inset: IInset) => void;
+
+export interface IInsetConfiguration {
+    /**
+     * A bit mask of InsetMask
+     * 
+     * @defaults DISPLAY_CUTOUT | SYSTEM_BARS
+     */
+    mask?: number;
+
+    /**
+     * If true, includes rounded corners in the inset information
+     * Only available on Android API 31 ("S") and later.
+     * 
+     * @defaults true
+     */
+    includeRoundedCorners?: boolean;
+}
+
+export class Inset { 
+    private static $instances: Map<string, Inset> = new Map();
+
+    private $currentInsets: IInset;
```

Comment:
plural naming (see all references to this private member)

Reply:
Also kept (but deprecated) `getInsets` in favour of `getInset()`.